src/hierarchy/Mammal.ts

```diff
@@ -7,7 +7,28 @@ export default abstract class Mammal extends Animal implements ICanJump {
     }
     
     
-    giveBirth(): void{
+    giveBirth(): [string, number, boolean,string][] {
+        if (this.isMale) {
+            throw Error(
+                `Only Females can give birth and ${this.name} is a proud male ${this.constructor.name}`
+            );
+        } else if (this.home === undefined) {
+            throw Error(
+                `${this.name} can't give birth because she doesn't have a home to meet a male`
+            );
+        } else {
+            this.timesGivenBirth += 1;
+            const candidatesforAnimals:[string, number, boolean,string][] = []
```

Comment:
this array of four magical values is used too often to be left like this, it has to be a class

Reply:
@teodora1203 suggested making it a type. I already made it this way. Should i change it?

Reply:
Type is also ok, might be a better version